src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceRepositoryTests.cs

```diff
@@ -0,0 +1,256 @@
+
+            var azureSubscriptions = await azureResourceRepository.DiscoverAzureSubscriptionsAsync();
+
+            Assert.NotStrictEqual(expectedSubscriptions, azureSubscriptions);
+
+            azureResourceGraph.Verify(m => m.QueryAzureLandscapeAsync("Discover Azure Subscriptions", It.IsAny<string>()), Times.Once);
+        }
+
+        [Fact]
+        public async Task AzureResourceGroups_Discovery_Succeeds()
+        {
+            var testData = ReadTestData("DiscoverAzureResourceGroupsAsync.json");
+            azureResourceGraph.Setup(m => m.QueryAzureLandscapeAsync("Discover Azure Resource Groups", It.IsAny<string>())).ReturnsAsync(testData);
+            var expectedResourceGroups = GetAzureResourceGroups();
+
+            var azureResourceGroups = await azureResourceRepository.DiscoverAzureResourceGroupsAsync();
+
+            Assert.NotStrictEqual(expectedResourceGroups, azureResourceGroups);
+
+            azureResourceGraph.Verify(m => m.QueryAzureLandscapeAsync("Discover Azure Resource Groups", It.IsAny<string>()), Times.Once);
+        }
+
+        private List<AzureSubscriptionInformation> GetAzureSubscriptions()
+        {
+            return new List<AzureSubscriptionInformation> {
+                 GetAzureSubscriptionInformation("c8819874-9e56-4e3f-b1a8-1c0325138f27",
+                                                 "0f9d7fea-99e8-4768-8672-06a28514f77e",
+                                                 "Visual Studio Enterprise",
+                                                 "Enabled",
+                                                 "Off",
+                                                 "Sponsored_2016-01-01",
+                                                 "RoleBased"),
+
+                 GetAzureSubscriptionInformation("c8819874-9e56-4e3f-b1a8-1c0325138f27",
+                                                 "0329dd2a-59dc-4493-aa54-cb01cb027dc2",
+                                                 "Windows Azure MSDN - Visual Studio Ultimate",
+                                                 "Enabled",
+                                                 "On",
+                                                 "MSDN_2014-09-01",
+                                                 "RoleBased")
+            };
+        }
+
+        private AzureSubscriptionInformation GetAzureSubscriptionInformation(string tenantId, string id, string name, string state, string spendingLimit, string quotaId, string authorizationSource)
+        {
+            return new AzureSubscriptionInformation
+            {
+                Id = id,
+                TenantId = tenantId,
+                Name = name,
+                AuthorizationSource = authorizationSource,
+                QuotaId = quotaId,
+                SpendingLimit = spendingLimit,
+                State = state
+            };
+        }
+
+        private List<AzureResourceGroupInformation> GetAzureResourceGroups()
```

Comment:
Is this based on the test data JSONs? If so, let's add a quick comment to clarify this

Reply:
Sure